osf/models/preprint_provider.py

```diff
@@ -46,6 +46,13 @@ def __unicode__(self):
         return '{} with id {}'.format(self.name, self.id)
 
     @property
+    def highlighted_subjects(self):
+        if self.subjects.filter(highlighted=True).exists():
+            return self.subjects.filter(highlighted=True).order_by('text')[:10]
+        else:
+            return sorted(self.top_level_subjects, key=lambda s: s.text)[:10]
```

Comment:
Will this work with if `self.top_level_subjects` returns a queryset instead of a list?